packages/foam-core/src/markdown-provider.ts

```diff
@@ -29,13 +29,13 @@ export function createNoteFromMarkdown(uri: string, markdown: string): Note {
     }
     return title === id ? CONTINUE : EXIT;
   });
-  const links: Link[] = [];
+  const links: NoteLink[] = [];
   visit(tree, node => {
     if (node.type === 'wikiLink') {
       links.push({
-        from: id,
         to: node.value as string,
         text: node.value as string,
+        position: node.position!
```

Comment:
Technically node.position may be null, but this should only be the case when the links have been generated/appended to the tree and not yet committed. The nodes here are always in markdown, and therefore should always have positions.